Model/Details.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ * Aligent Consulting
+ * Copyright (c) Aligent Consulting (https://www.aligent.com.au)
+ */
+
+declare(strict_types=1);
+
+namespace Aligent\AsyncEvents\Model;
+
+use Aligent\AsyncEvents\Api\AsyncEventRepositoryInterface;
+use Aligent\AsyncEvents\Model\ResourceModel\AsyncEventLog\Collection;
+use Aligent\AsyncEvents\Model\ResourceModel\AsyncEventLog\CollectionFactory as AsyncEventLogCollectionFactory;
+
+class Details
+{
+    /**
+     * @var array
+     */
+    private $traceCache = [];
+
+    /**
+     * @var Collection
+     */
+    private $collection;
+
+    /**
+     * @var AsyncEventRepositoryInterface
+     */
+    private $asyncEventRepository;
+
+    public function __construct(
+        AsyncEventLogCollectionFactory $collectionFactory,
+        AsyncEventRepositoryInterface  $asyncEventRepository
+    ) {
+        $this->collection = $collectionFactory->create();
+        $this->asyncEventRepository = $asyncEventRepository;
+    }
+
+    public function getDetails($uuid)
```

Comment:
Will this ever be called twice in the same request? If so, you should be creating the collection here, rather than the constructor.